selene-lib/src/lints/global_usage.rs

```diff
@@ -1,23 +1,43 @@
 use super::*;
-use std::{collections::HashSet, convert::Infallible};
+use std::collections::HashSet;
 
 use full_moon::ast::Ast;
+use regex::Regex;
+use serde::Deserialize;
 
 fn is_global(name: &str, roblox: bool) -> bool {
     (roblox && name == "shared") || name == "_G"
 }
 
-pub struct GlobalLint;
+#[derive(Clone, Deserialize)]
+#[serde(default)]
+pub struct GlobalConfig {
+    ignore_pattern: String,

```

Comment:
Instead of doing `.^`, we should instead make this `Option<String>` and save the performance IMO.